sched/scheduler/stateful_scheduler.go

```diff
@@ -92,11 +105,11 @@ func NewStatefulScheduler(
 
 	// TODO: we need to allow the scheduler to accept new jobs
 	// while recovering old ones.
-	if recoverJobsOnStartup {
+	if config.RecoverJobsOnStartup {
 		sched.startUp()
 	}
 
-	if !debugMode {
+	if !config.DebugMode {
```

Comment:
This seems like it will be hard to use efficiently in practice. I'm not sure this is useful in production without a mechanism to specify which jobs are to be manually advanced, or a way to block jobs that are spam wrt debugging. So I would say this is fine but incomplete.